decidim-core/lib/decidim/core/test/factories.rb

```diff
@@ -9,8 +9,39 @@
 require "decidim/assemblies/test/factories"
 require "decidim/comments/test/factories"
 
-def generate_localized_title
-  Decidim::Faker::Localized.localized { generate(:title) }
+def generate_component_name(locales, manifest_name, skip_injection: false)
+  prepend = skip_injection ? "" : "<script>alert(\"#{manifest_name}\");</script>"
+
+  Decidim::Components::Namer.new(locales, manifest_name).i18n_name.transform_values { |v| [prepend, v].compact_blank.join(" ") }
+end
+
+def generate_localized_description(field = nil, skip_injection: false, before: "<p>", after: "</p>")
+  Decidim::Faker::Localized.wrapped(before, after) do
+    generate_localized_title(field, skip_injection:)
+  end
+end
+
+def generate_localized_word(field = nil, skip_injection: false)
+  skip_injection = true if field.nil?
+  Decidim::Faker::Localized.localized do
+    if skip_injection
+      Faker::Lorem.word
+    else
+      "<script>alert(\"#{field}\");</script> #{Faker::Lorem.word}"
```

Comment:
Wow having the field in the alert is a great idea, really clever! Makes easier to find out where is the injection 👏🏽

Reply:
Closing, as there is nothing to be fixed here.